src/http/index.ts

```diff
@@ -18,16 +18,22 @@ export interface HttpOptions {
   routesImporter?: RoutesImporter;
 
   // An array of valid express ApplicationRequestHandlers (middlewares) injected BEFORE loading routes
-  preRouteApplicationRequestHandlers?: any | [string, any][];
+  requestMiddleware?: any | [string, any][];
 
   // an array of valid express ApplicationRequestHandlers (middlewares) injected AFTER loading routes
-  postRouteApplicationRequestHandlers?: any | [string, any][];
+  errorMiddleware?: any | [string, any][];
+
+  // optional error hook function on application error
+  errorHook?: (error: any) => void;
+
+  // optional error logger which replaces console.error on application error
+  errorLogger?: (error: any) => void;
 }
 
-export default async (port: number, options?: HttpOptions): Promise<Http> => {
+export default async (port: number, options: HttpOptions = {}): Promise<Http> => {
```

Comment:
defaulting is fine, but assuming that `null` was not passed in can lead to exceptions within the exception handler (or this file).

it's a bit more clear to leave it unspecified, then the usual safe-navigation checks still fly. If it looks like it will always be an object it might lead to people assuming that there is no need to `options?.something`

Reply:
personally not a fan of the ? operator but i don't care too much eitherway happy to go ?